### DISCUSSION TITLE
Higher accuracy for raycasts

### DISCUSSION THREAD

**Discussion by DocAce** [jrouwe/JoltPhysics#190] [Q&A] [answered]
We are almost certainly going to need to be able to do raycasts against detailed meshes, which we obviously can't (and don't want) to use for collision detection. I have currently solved this by giving each object that needs it a second body on a special ObjectLayer, which is always excluded in MyObjectCanCollide() and MyBroadPhaseCanCollide(). Raycasts can specify whether they want to hit these detailed objects or the regular ones which is just as easily done with an ObjectLayerFilter.

Is this actually a good solution or can anyone think of something better? The other solutions I can think of are:
* have the meshes in the main physics body alongside the convex shapes and somehow exclude them from collision detection (maybe with a ShapeFilter)
* create a separate physics world for raycasting (I strongly suspect the performance benefits would be negligible to nonexistent)

**Comment by jrouwe** [comment, marked as answer]
> I have currently solved this by giving each object that needs it a second body

This is exactly how we're solving it. This second body is not even a keyframed object in our case, we just set its position every time the first body moves.
